### ISSUE TITLE
Unclosed string breaks `overlay use`

### ISSUE THREAD

**Issue by pixilcode** [nushell/nushell#11254]
### Describe the bug

When the overlay file that is being used has a string that is unclosed, `overlay use` waits for more input.

### How to reproduce

1. Create a file, `foo.nu` with the following contents:
   ```
   const value = "This string is unclosed
   ```
2. Use the file as an overlay with `overlay use foo.nu`

### Expected behavior

When I run `overlay use foo.nu`, I expect to get an error message saying that the string is unclosed. However, instead it just prompts for more input:
```
~> overlay use foo.nu
::: 
::: 
::: 
```
Only Ctrl-C will close it, but it won't add the overlay.

### Screenshots

![image](https://github.com/nushell/nushell/assets/22626010/1e7ffcd1-1fab-4dff-be27-44a62163db69)

### Configuration

```
~> version | transpose key value | to md --pretty
| key                | value                                     |
| ------------------ | ----------------------------------------- |
| version            | 0.87.0                                    |
| branch             |                                           |
| commit_hash        |                                           |
| build_os           | linux-x86_64                              |
| build_target       | x86_64-unknown-linux-gnu                  |
| rust_version       | rustc 1.73.0 (cc66ad468 2023-10-03)       |
| rust_channel       | stable-x86_64-unknown-linux-gnu           |
| cargo_version      | cargo 1.73.0 (9c4383fb5 2023-08-26)       |
| build_time         | 2023-11-16 17:30:33 -07:00                |
| build_rust_channel | release                                   |
| allocator          | mimalloc                                  |
| features           | default, extra, sqlite, trash, which, zip |
| installed_plugins  |                                           |
```

**Comment by amtoine** [comment]
yeah i've seen that with `use` and `source` as well, imo it should throw syntax errors :+1: 

i wouldnt' be surprised if we already had issues on that topic

**Comment by ysthakur** [comment]
I'm not 100% sure that this is what's happening, but I think that when you try to parse `source foo.nu`, `overlay use foo.nu`, etc., [it parses `foo.nu`](https://github.com/nushell/nushell/blob/main/crates/nu-parser/src/parse_keywords.rs#L3380). So when you type `overlay use foo.nu` or other similar commands, it will try to parse `foo.nu` and add an `UnexpectedEof` error to the working set. When Reedline tries to validate the line, it will see that there's an `UnexpectedEof` error and prompt for more input.

If this is really what's happening, perhaps we should delay parsing `foo.nu` until the command is actually evaluated, but maybe there's a reason for doing it right when it's parsed?
